### DISCUSSION TITLE
TechNews - Rendering process failed

### DISCUSSION THREAD

**Discussion by kennyheimbuch** [GetPublii/Publii#1469] [Help]
Version: 0.45.2 (build 16609), TechNews 2.7.0.0, Windows 10.

Trying to build site, I get this error:

Rendering process failed
An error (1002) occurred during parsing index.hbs file.
Missing helper: "getTagColor"

Any ideas how to fix this or what the cause is?

Edit: have answered my own question, digging around. Having an override folder from the previous TechNews theme (2.6.1.0), caused a parsing error due to the tag color parsing routine changed in the new theme version.

**Comment by iffinland** [comment]
I don't understand exactly, but did you change the TAG colors yourself in the settings?

I also tried changing the default TAG colors to something else and I didn't get any errors with rendering and publishing, everything works fine.

You can try to change the theme for a while and then back again later, maybe some momentary problem. Sorry if I don't understand your problem

![Screenshot 2024-04-06 07 56 51](https://github.com/GetPublii/Publii/assets/157421268/0e4753dd-f3ad-4b52-9f09-c6e4c7674569)

**Comment by kennyheimbuch** [comment]
What happened is that I was using an override for the Technews theme that had some edits to index.hbs.  That was an older version of the newer theme's version of that file. Version 2.7.0.0 doesn't have some routine for handling tags that the last version had, so parsing failed. (v 2.7.0.0 was the first version to have the new tag color process). 

The rendering process failed when I installed the new theme, so I never had the chance to set the tag colors in the new theme version.

Example, the getTagColor routine is different:

Version 2.6.1.0
```
                        {{#if @config.custom.cardsTag}}
                           {{#if mainTag}}
                              <a href="{{mainTag.url}}" class="{{#featuredImage}}
                                           {{#if url}}c-card__tag{{/if}}{{/featuredImage}} u-tag {{ getTagColor mainTag.slug @config.custom.tagColors}}">{{mainTag.name}}</a>
                           {{/if}}
```

Version 2.7.0.0
```
                        {{#if @config.custom.cardsTag}}
                           {{#if mainTag}}
                              <a href="{{mainTag.url}}" class="{{#featuredImage}}
                                           {{#if url}}c-card__tag{{/if}}{{/featuredImage}} u-tag u-tag--{{mainTag.slug}}">{{mainTag.name}}</a>
                           {{/if}}
```

**Comment by bobmitro** [comment]
The handling of tag coloring has been updated in the theme's v 2.7.0.0. So, you will need to merge the overridden files or add the missing helper.js (from the previous theme version) to the root of the overridden folder.

**Reply by kennyheimbuch** [reply]
Thanks Bob! (that new tags handling is a nice feature!)
